addons/mrp/models/mrp_production.py

```diff
@@ -200,6 +200,12 @@ def _get_default_location_dest_id(self):
     production_location_id = fields.Many2one('stock.location', "Production Location", related='product_id.property_stock_production', readonly=False)
     picking_ids = fields.Many2many('stock.picking', compute='_compute_picking_ids', string='Picking associated to this manufacturing order')
     delivery_count = fields.Integer(string='Delivery Orders', compute='_compute_picking_ids')
+    is_ready_produce = fields.Boolean(compute='_compute_is_ready_produce', default=False)
+
+    @api.depends('move_raw_ids', 'move_raw_ids.reserved_availability')
+    def _compute_is_ready_produce(self):
+        for order in self:
+            order.is_ready_produce = all(move.bom_line_id.product_qty <= move.reserved_availability for move in order.move_raw_ids) 
```

Comment:
becarefull `all([]) = True`

why use `bom_line_id.product_qty`; you can modifiy (or delete) the value of quantity or the unit on the bom line, just use `move.product_uom_qty`, no ?

avoid useless compute if it is an done or cancel order
```python
for order in self.filtered(lambda x: x.state not in ('xxxx')):
    order.is_ready_produce = all(move.product_qty <= move.reserved_availability for move in order.move_raw_ids)
```

Reply:
all([]) is not possible in Confirmed state because manufacturing order always have raw materials in Confirmed state
here , we don't want to modify or delete the value of quantity.
we are just comparing the move.bom_line_id.product_qty and move.reserved_availability to know whether reserved_availability is enough.

Reply:
`move.bom_line_id.product_qty` is not a fixed value, you can lauch a manufacturing order, then modifiy the bom, and your compute will be wrong, it think you need to use `move.product_qty` or`move.product_uom_qty` , check what is made on `stock.picking`.

Reply:
Your case is right ( if some one change current production bom ) ... but  we can not use 
move.product_qty <= move.reserved_availability. also reserve quantity will not greater
 in any case  then move quantity

We can use unit_factor <= reservered ( factor will not change when bom change )

Reply:
you can also delete a bom line, after lauching MO. That why it is not relevant to use bom_line, it is better to user the state of the move.